### ISSUE TITLE
Shared libraries: libpulse.so.0

### ISSUE THREAD

**Issue by hamorillo** [mindrunner/docker-android-sdk#44]
I'm trying to create a new emulator. I've downloaded with the SdkManager the needed packages but when I execute the command to create de virtual device I get the following error.

> /opt/android-sdk-linux/emulator/qemu/linux-x86_64/qemu-system-x86_64: error while loading shared libraries: libpulse.so.0: cannot open shared object file: No such file or directory

Do you know what is happening?

**Comment by mindrunner** [comment]
Please give a reproducible step by step example of your commands.

**Comment by hamorillo** [comment]
I'm executing the following steps:

`yes | ${ANDROID_SDK}/tools/bin/sdkmanager --licenses`
 `${ANDROID_SDK}/tools/bin/sdkmanager 'system-images;android-28;google_apis;x86'`
    `${ANDROID_SDK}/tools/bin/avdmanager create avd --force -n device --package 'system-images;android-28;google_apis;x86' --device "Nexus 5X" -c 512M`
    `${ANDROID_SDK}/emulator/emulator -port 5558 @device &`

**Comment by mindrunner** [comment]
It seems like the alpine version lacks some libraries. pulse and others need to be installed. I can't believe I missed out on this. If you switch to the ubuntu variants, it should be fine.

Also feel free to find the correct libraries for alpine and we can integrate them into the image. You definitely need the following:

- pulseaudio-libs
- mesa-gl
- fontconfig

but there are more missing. I am pretty sure, there is a doc out there telling us exactly what we need. :)

**Comment by oxr463** [comment]
@hamorillo please test again so we can close this issue.

**Comment by mindrunner** [comment]
`docker run --rm -ti runmymind/docker-android-sdk:alpine-standalone bash`
`${ANDROID_SDK}/tools/bin/sdkmanager 'system-images;android-28;google_apis;x86'`
`${ANDROID_SDK}/tools/bin/avdmanager create avd --force -n device --package 'system-images;android-28;google_apis;x86' --device "Nexus 5X" -c 512M`
`${ANDROID_SDK}/emulator/emulator -port 5558 @device`
--->
`/opt/android-sdk-linux/emulator/qemu/linux-x86_64/qemu-system-x86_64: error while loading shared libraries: libXcursor.so.1: cannot open shared object file: No such file or directory`

> but there are more missing. I am pretty sure, there is a doc out there telling us exactly what we need. :)

I think we need to find all the missing libraries and add them manually. Google could probably help, otherwise trial and error.

**Comment by NinjaRat84** [comment]
I have tried to trigger the emulator like @hamorillo :

```
# Add support for libxcursor and libasound
- apk add libxcursor
- apk add alsa-lib

# Create and start the emulator
- echo y | $ANDROID_HOME/tools/bin/sdkmanager "system-images;android-30;google_apis;x86_64"
- echo no | $ANDROID_HOME/tools/bin/avdmanager create avd -n test --force -k "system-images;android-30;google_apis;x86_64" --abi google_apis/x86_64
    - $ANDROID_HOME/emulator/emulator -avd test -skin 768x1280 -netdelay none -no-snapshot -wipe-data -no-audio -no-boot-anim &

# Wait for the device to start and catch it
- $ANDROID_HOME/platform-tools/adb wait-for-device
```

But it's doesn't seem to fix it in the runner instance:

```
$ $ANDROID_HOME/platform-tools/adb wait-for-device
* daemon not running; starting now at tcp:5037
* daemon started successfully
/opt/android-sdk-linux/emulator/qemu/linux-x86_64/qemu-system-x86_64: /usr/lib/libasound.so.2: no version information available (required by /opt/android-sdk-linux/emulator/lib64/qt/lib/libQt5WebEngineCoreAndroidEmu.so.5)
/opt/android-sdk-linux/emulator/qemu/linux-x86_64/qemu-system-x86_64: Relink `/usr/lib/libogg.so.0' with `/usr/glibc-compat/lib/libc.so.6' for IFUNC symbol `memmove'
```

Any hints on how to proceed from here?

**Comment by oxr463** [comment]
@NinjaRat84 see: https://github.com/sgerrand/alpine-pkg-glibc/issues/58

**Comment by mindrunner** [comment]
Thanks for the links as a reference. I am aware of those issues and afaik there is no proper stable solution yet. However, I am happy to see a MR with an experimantal alpine android image with working emulator! :)

**Comment by mindrunner** [comment]
@NinjaRat84 are you sure, you don't want to just use the ubuntu edition of this image? You won't face those issues there.

**Comment by NinjaRat84** [comment]
@mindrunner Yes, I tried to switch to the ubuntu-standalone version and that issue was removed.

However, another emerged:

`emulator: ERROR: Can't find 'Linux version ' string in kernel image file: /opt/android/system-images/android-30/google_apis/x86_64//kernel-ranchu`

Doesn't matter if its running in a kubernetes cluster or on an actual gitlab-runner on a local unubtu 18 computer.
It does however work as intended when running on my local runner on Mac OSX.

Any suggestions on how to proceed?

**Comment by mindrunner** [comment]
I can only find this:
https://stackoverflow.com/questions/60319486/android-emulator-api-24-cant-start-due-to-kernel-image-file-kernel-ranchu

To investigate, I need proper information on how to reproduce that. Also, it is pretty off-topic in this ticket. So feel free to create a ticket for this, so that we do not pollute this one with a different topic.

**Comment by mindrunner** [comment]
Closing this, due to staleness. Solution is to use the ubuntu flavor.
